app/src/main/java/com/arielwang/workoutlogger/features/workoutaddingflow/track/ui/screen/TrackViewModel.kt

```diff
@@ -188,38 +197,47 @@ class TrackViewModel @Inject constructor(
                 )
                 emitViewState()
             }
+            is TrackView.Action.ChangeShouldShowMaxCharState -> {
+                viewState = viewState.copy(
+                    shouldShowMaxCharAlert = ShouldShowMaxCharAlert(shouldShow = action.shouldShow)
+                )
+                emitViewState()
+            }
         }
     }
 
     private fun emitViewState() {
         _uiState.value = viewState
     }
 
-    // the value of TextField cannot be negative
-    private fun noNegativeNumber(text: String): String {
-        return if ((text.toInt() - 1) < 0) {
-            "0"
-        } else {
-            (text.toInt() - 1).toString()
-        }
+    // the minimum value of the TextField is zero
+    private fun coerceAtLeastZero(number: Int): Int {
```

Comment:
Not sure if this is worth wrapping as a function. You can use coerceAtLeast directly instead

Reply:
Thank you.